### PR TITLE
proof-of-concept(v2): docusaurus-lint-api

### PR DESCRIPTION
**Pull request by fanny**
## Motivation
As discussed in #2965, it would be good to have a linter for our docs

From what I saw, there are a couple of styleguide refereces, but it are limited to webpages, like this: https://developers.google.com/tech-writing/two/large-docs

Having a tool that make this, save time and effort, we see this when we use eslint in our projects.
This PR adds a POC for the proposal, I'm using the [`markdown-lint`](https://github.com/DavidAnson/markdownlint), if you have interest in see how the lib parses the tokens, [check the parser docs](https://github.com/markdown-it/markdown-it/blob/master/docs/architecture.md)

## Test Plan

go to docusaurus templates, `bootstrap/`
run `yarn lint`
see if the output returned is this:
```
Results {
  '/Users/fannyvieira/Developing/docusaurus/packages/docusaurus-plugin-lint-api/src/__tests__/test.md': [
    {
      lineNumber: 3,
      ruleNames: [Array],
      ruleDescription: 'Heading levels should only increment by one level at a time',
      ruleInformation: 'https://github.com/DavidAnson/markdownlint/blob/v0.20.3/doc/Rules.md#md001',
      errorDetail: 'Expected: h2; Actual: h3',
      errorContext: null,
      errorRange: null
    },
    {
      lineNumber: 5,
      ruleNames: [Array],
      ruleDescription: 'Line length',
      ruleInformation: 'https://github.com/DavidAnson/markdownlint/blob/v0.20.3/doc/Rules.md#md013',
      errorDetail: 'Expected: 80; Actual: 178',
      errorContext: null,
      errorRange: [Array]
    },
    {
      lineNumber: 15,
      ruleNames: [Array],
      ruleDescription: 'Line length',
      ruleInformation: 'https://github.com/DavidAnson/markdownlint/blob/v0.20.3/doc/Rules.md#md013',
      errorDetail: 'Expected: 80; Actual: 91',
      errorContext: null,
      errorRange: [Array]
    },
    {
      lineNumber: 25,
      ruleNames: [Array],
      ruleDescription: 'Line length',
      ruleInformation: 'https://github.com/DavidAnson/markdownlint/blob/v0.20.3/doc/Rules.md#md013',
      errorDetail: 'Expected: 80; Actual: 98',
      errorContext: null,
      errorRange: [Array]
    },
    {
      lineNumber: 33,
      ruleNames: [Array],
      ruleDescription: 'Line length',
      ruleInformation: 'https://github.com/DavidAnson/markdownlint/blob/v0.20.3/doc/Rules.md#md013',
      errorDetail: 'Expected: 80; Actual: 195',
      errorContext: null,
      errorRange: [Array]
    },
    {
      lineNumber: 43,
      ruleNames: [Array],
      ruleDescription: 'Line length',
      ruleInformation: 'https://github.com/DavidAnson/markdownlint/blob/v0.20.3/doc/Rules.md#md013',
      errorDetail: 'Expected: 80; Actual: 110',
      errorContext: null,
      errorRange: [Array]
    }
  ]
}
✨  Done in 2.60s.
```
**I'll better format the output**
## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: packages/docusaurus-plugin-lint-api/package.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "name": "@docusaurus/plugin-lint-api",
+  "version": "2.0.0-alpha.58",
+  "description": "A lint for API docs",
+  "main": "src/index.js",
+  "license": "MIT",
+  "devDependencies": {
+    "markdownlint": "^0.20.3"
+  },
+  "peerDependencies": {
+    "markdown-it": "^10.0.0"
```
Comment (slorber) [inline packages/docusaurus-plugin-lint-api/package.json:11]: requiring the user to add the deps is likely to be annoying
both should rather be "dependencies" as it's code the end user will run

**Comment by slorber** [comment]
Some ideas:
- create some custom rules, expose them as a Docusaurus package
- use `.markdownlint.json` in the templates, use the custom rules by default, and run them as a npm script 

Not 100% sure there's value integrating this with the docusaurus cli  compared to leverarging the existing markdownlint-cli (no strong opinion, just some thoughts)

**Comment by fanny** [comment]
I agree with all of your points, about why not use the `markdownlint-cli`, I think if we do this using docusaurus, we'll have more visibility, but I can develop as my repo, and put as a showcase. WDYT?

**Comment by fanny** [comment]
I've decided to create a separate repo as you suggested, I'm finishing the rules.

**Comment by slorber** [comment]
great :) tell me when you have something to show

**Pull request closed without merging**